valkyrie/lib/valkyrie/storage/memory.rb

```diff
@@ -27,5 +27,31 @@ def find_by(id:)
     def handles?(id:)
       id.to_s.start_with?("memory://")
     end
+
+    # @param id [Valkyre::ID]
+    # @param digests [Digest]
+    # @return [Digest]
+    def checksum(id:, digests:)
```

Comment:
I wonder if this and valid? should be part of the interface for Valkyrie::Storage::File instead.

Reply:
Maybe? The API is a little cumbersome, but having it here means that the implementation doesn't have to figure out which StorageAdapter to use.

Reply:
I think we should be open to different adapters returning different classes(with the same interface) for find_by, to get by that complication.

Reply:
Sure, but what I mean is that it's fine for an app to make assumptions about which StorageAdapter a file is in (e.g., https://github.com/samvera-labs/valkyrie/pull/219/files#diff-2cb4c26e6599735014380839f6f2fc21R28), but I wouldn't want to make that kind of assumption in the gem.

But if we move `checksum` and `valid?` to `Valkyrie::Storage::File`, then how would it know which StorageAdapter to use?  Passing it as a parameter seems backwards.  So is the alternative checking in every StorageAdapter?

Reply:
The IDs for a file are unique to the storage adapter it's for. What I would expect to work is something like:

```ruby
file = Valkyrie::StorageAdapter.find_by(id: my_id)
file.valid?(size: size, digests: {sha256: checksum})
```

Am I still misunderstanding?

Reply:
The reason I'm inclined to put `valid?` and `checksum` at the layer of the returned file is because we pass the ID, but just so we can pass it into `find_by`. Seems like a code smell.

Reply:
Oh, since the file IDs are unique, then we can safely just go find a StorageAdapter that has it — OK, I'll do that.

Reply:
Just pushed a new commit to do this — turns out it was much simpler because all the info needed for the fixity checking was already available anyway.  As I mentioned in #216, there are some shared_specs updates that are still needed.